### PR TITLE
Migrate from VMware implementation

### PR DESCRIPTION
**Pull request by scothis**
Hooks to migrate from the VMware implementation are available opt-in by
passing the --migrate-from-vmware flag to the manager container. Running
with the migration hooks installed will incur a performance penalty and
should only be used while a migration is active, and then disabled.

Depends on #327

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,7 @@ help: ## Display this help.
 .PHONY: manifests
 manifests: ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	@rm -rf config/crd/bases/_.yaml
```
Comment (sadlerap) [inline Makefile:56]: Q: why is this necessary?  This file doesn't exist for me when I build from a clean repository.

Comment (scothis) [reply Makefile:56]: I think it's a bug in controller-gen triggered by the PodSpecable duck type resource which is like an object but shouldn't generate a CRD. I can try to hunt the origin another time, nuking the output seemed like the easiest path forward.

**Review by sadlerap — APPROVED** [review]
lgtm, just waiting on ci
